### PR TITLE
[MOF] Begin formal JSON schema

### LINKED ISSUE #46: Consider a JSON schema
https://json-schema.org/

This would be a solid step towards standardizing the format and ensuring that it's not implementation dependent.

### PR DESCRIPTION
**Pull request by odow**
Still todo
- [x] <s>is there a way to validate variable names used in functions are defined in `variables`?</s> Don't think so
- [x] nonlinear functions
- [x] variable and constraint attributes other than `Name`
- [x] rebase on #49 once merged

Closes #46

### CODE REVIEW

File: src/MOF/mof.schema.json [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+    "type": "object",
+    "properties": {
+        "version": {
+            "const": 0
+        },
+        "author": {
+            "description": "The author of the model for citation purposes",
+            "type": "string"
+        },
+        "description": {
```
Comment (mlubin) [inline src/MOF/mof.schema.json]: Is there a name field on the model? I.e. corresponding to http://www.juliaopt.org/MathOptInterface.jl/dev/apireference/#MathOptInterface.Name.

**Comment by odow** [comment]
Yes, it's very easy to read. I wonder if we should include the math descriptions for all functions and sets in the schema so that it is virtually self-documenting.

File: src/MOF/schema/examples/quadratic.mof.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+    "description": "The problem: min{x^2 + x * y + y^2}",
```
Comment (Wikunia) [inline src/MOF/schema/examples/quadratic.mof.json]: Based on 
```
{
     "coefficient": 2,
     "variable_1": "x",
     "variable_2": "y"
 }
```
it should be `min{x^2 + 2 * x * y + y^2}`, right?

Comment (odow) [reply src/MOF/schema/examples/quadratic.mof.json]: Oops. You're right that there is a mistake but that is the wrong one. I doubled the wrong value.

This is an annoying point that has come up a few times (e.g., https://github.com/JuliaOpt/LinQuadOptInterface.jl/issues/22).

MOI defines the [ScalarQuadraticFunction](http://www.juliaopt.org/MathOptInterface.jl/v0.1/apireference.html#MathOptInterface.ScalarQuadraticFunction) as `0.5 x' Q x`.

**Comment by odow** [comment]
Merging since I think we're fairly happy with this. We can address any other points in future PR's.

**Pull request merged**